src/directives/offense/poisonRoom.ts

```diff
@@ -1,169 +1,188 @@
-// 				if(this.room.roads.length > 0){
-// 					_.forEach(this.room.roads, road => {road.destroy();} );
-// 				}
-// 				this.room.controller!.unclaim();
-// 				//log.notify(`Removing poisonRoom directive in ${this.pos.roomName}: operation completed.`);
-// 				//this.remove();
-// 			} else {
-// 				this.poison();
-// 			}
-// 		}
-//
-// 		// Remove if owned by other player
-// 		if (Game.time % 10 == 2 && this.room && !!this.room.owner && this.room.owner != MY_USERNAME) {
-// 			log.notify(`Removing poisonRoom directive in ${this.pos.roomName}: room already owned by another player.`);
-// 			this.remove();
-// 		}
-// 	}
-// }
-//
-//
-//
-//
-//
+import {Colony, ColonyMemory, getAllColonies} from '../../Colony';
+import {log} from '../../console/log';
+import {ClaimingOverlord} from '../../overlords/colonization/claimer';
+import {RoomPoisonerOverlord} from '../../overlords/offense/roomPoisoner';
+import {StationaryScoutOverlord} from '../../overlords/scouting/stationary';
+import {profile} from '../../profiler/decorator';
+import {Cartographer, ROOMTYPE_CONTROLLER} from '../../utilities/Cartographer';
+import {printRoomName} from '../../utilities/utils';
+import {MY_USERNAME} from '../../~settings';
+import {DirectiveOutpostDefense} from '../defense/outpostDefense';
+import {Directive} from '../Directive';
+import {DirectiveControllerAttack} from './controllerAttack';
+
+export const RUN_TIMER = 25;
+
+/*
+* Poison sources in remote rooms by claiming controller, walling in its sources and controller then unclaiming it.
+*/
+@profile
+export class DirectivePoisonRoom extends Directive {
+	static directiveName = 'PoisonRoom';
+	static color = COLOR_RED;
+	static secondaryColor = COLOR_BROWN;
+	static requiredRCL = 4;
+	walkableSourcePosisions: RoomPosition[];
+	walkableControllerPosisions: RoomPosition[];
+	overlords: {
+		claim: ClaimingOverlord;
+		roomPoisoner: RoomPoisonerOverlord;
+	};
+	constructor(flag: Flag) {
+		super(flag, colony => colony.level >= DirectivePoisonRoom.requiredRCL);
+		// Remove if misplaced
+		if (Cartographer.roomType(this.pos.roomName) != ROOMTYPE_CONTROLLER) {
+			log.warning(`${this.print}: ${printRoomName(this.pos.roomName)} is not a controller room; ` +
+						`removing directive!`);
+			this.remove(true);
+		}
+		// failsafe - if already owned and controller level > 2, remove flag
+		if(this.room && this.room.controller && this.room.controller.my && this.room.controller.level > 2) {
+			this.remove(true);
+		}
+		// Remove if owned by other player
+		if (this.room && !!this.room.owner && this.room.owner != MY_USERNAME) {
+			log.notify(`Removing poisonRoom directive in ${this.pos.roomName}: room already owned by another player.`);
+			this.remove(true);
+		}
+		// Remove if not enough GCL
+		if(getAllColonies().length == Game.gcl.level) {
+			log.warning(`${this.print}: ${printRoomName(this.pos.roomName)} not enough GCL; ` +
+						`for poison room, removing directive!`);
```

Comment:
looks like you forgot to include `this.remove()`

Reply:
I removed it all together, because once I claim to poison it might return true and the directive will get removed